From ea31df4e70600b983fd4c21c07b18cc3d48dd0d7 Mon Sep 17 00:00:00 2001
From: ruslanbay <67730802+ruslanbay@users.noreply.github.com>
Date: Fri, 28 Nov 2025 13:50:53 +0100
Subject: [PATCH 10/16] fix: rename acpi_bus_get_device to acpi_fetch_acpi_dev
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

drivers/media/platform/intel/ipu4-acpi.c: In function ‘get_acpi_dep_data’:
drivers/media/platform/intel/ipu4-acpi.c:482:21: error: implicit declaration of function ‘acpi_bus_get_device’; did you mean ‘pci_bus_add_device’? [-Werror=implicit-function-declaration]
  482 |                 if (acpi_bus_get_device(dep_devices.handles[i], &device))
      |                     ^~~~~~~~~~~~~~~~~~~
      |                     pci_bus_add_device
---
 drivers/media/platform/intel/ipu4-acpi.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/media/platform/intel/ipu4-acpi.c b/drivers/media/platform/intel/ipu4-acpi.c
index 4db674e26..bf9bdcb80 100644
--- a/drivers/media/platform/intel/ipu4-acpi.c
+++ b/drivers/media/platform/intel/ipu4-acpi.c
@@ -479,7 +479,8 @@ static int get_acpi_dep_data(struct device *dev,
 			continue;
 
 		/* Process device IN3472 created by acpi */
-		if (acpi_bus_get_device(dep_devices.handles[i], &device))
+		device = acpi_fetch_acpi_dev(dep_devices.handles[i]);
+		if (!device)
 			return -ENODEV;
 
 		dev_dbg(dev, "Depend ACPI device found: %s\n",
-- 
2.34.1

